src/components/Sidebar/Sidebar.tsx

```diff
@@ -0,0 +1,218 @@
+import {
+  Box,
+  Divider,
+  Text,
+  VStack,
+  Tabs,
+  TabList,
+  Tab,
+  Icon,
+  HStack,
+  Flex,
+  Skeleton,
+  Spacer,
+} from "@chakra-ui/react"
+import { Button, ButtonProps } from "@opengovsg/design-system-react"
+import { PropsWithChildren } from "react"
+import {
+  BiBook,
+  BiBuoy,
+  BiCog,
+  BiCubeAlt,
+  BiFile,
+  BiGridAlt,
+  BiImage,
+  BiLogOutCircle,
+} from "react-icons/bi"
+import { useParams, Link as RouterLink } from "react-router-dom"
+
+import { LOCAL_STORAGE_KEYS } from "constants/localStorage"
+
+import { useLastUpdated } from "hooks/useLastUpdated"
+import { useLocalStorage } from "hooks/useLocalStorage"
+import { useSidebarContext } from "hooks/useSidebar"
+
+import { TabSection } from "types/sidebar"
+
+const getTabIndex = (section: TabSection) => {
+  switch (section) {
+    case "workspace":
+      return 0
+    case "resourceRoom":
+      return 1
+    case "images":
+      return 2
+    case "files":
+      return 3
+    case "settings":
+      return 4
+    default: {
+      const exception: never = section
+      throw new Error(exception)
+    }
+  }
+}
+
+export const Sidebar = (): JSX.Element => {
+  const { siteName } = useParams<{ siteName: string }>()
+  const { lastUpdated, isError, isLoading } = useLastUpdated(siteName)
+  const { selectedTab, setSelectedTab } = useSidebarContext()
+  const [loggedInUser] = useLocalStorage(
+    LOCAL_STORAGE_KEYS.GithubId,
+    "Unknown user"
+  )
+
+  return (
+    <Flex bg="white" h="100vh" flexDir="column" w="15rem">
+      <Box p="1.5rem">
+        <VStack align="flex-start" spacing="0.5rem">
```

Comment:
The sidebar still scrolls on smaller screens - might need to modify the height to account for the header (and possibly the banner)

Reply:
changed, but only accounts for header. for banner, i'd need to check to see if multiline banners are a possibility in which case, it's probably unfeasible to do (need to dynamically calc at runtime the height of the banner element)

Reply:
This seems somewhat fixed now, but I'm not sure if this issue is related:

https://user-images.githubusercontent.com/22111124/173554086-1b528e36-cafe-4878-939c-ecb2a322d0b4.mov

In the above clip, it looks like there's an inner and outer scroll bar - this is causing the scrolling to snap to the end of one scroll bar, causing it to be less smooth overall. I don't think it's a huge issue, but it would be good if this problem had a quick fix

Reply:
@alexanderleegs i can fix this actually but this is super confusing because the reason the scrollbar appears is cos there's overflow (as `overflow: auto` is set). 

**however**, the overflow is NOT actually anything?! when i set overflow to visible, the overflow is actually just a grayed out area. i'm not sure what's happening here but we don't seem to be losing anything so technically this is fixable just by doing `overflow: hidden` 

idk why the overflow is just blank space tho ._.

Reply:
Sure, this looks great - remember to implement this for the remaining refactors later as well